test/util.cpp

```diff
@@ -127,49 +153,69 @@ bool operator==(const ProjectedRing& a, const ProjectedRing& b) {
     EXPECT_EQ(a.points, b.points);
     return true;
 }
+*/
 
-std::vector<TileFeature>
+mapbox::geometry::feature_collection<int16_t>
 parseJSONTile(const rapidjson::GenericValue<rapidjson::UTF8<>, rapidjson::CrtAllocator>& tile) {
-    std::vector<TileFeature> features;
+    mapbox::geometry::feature_collection<int16_t> features;
     EXPECT_TRUE(tile.IsArray());
     for (rapidjson::SizeType k = 0; k < tile.Size(); ++k) {
         const auto& feature = tile[k];
 
+        /*
         TileFeatureType tileType = TileFeatureType::Point;
         if (feature.HasMember("type")) {
             const auto& type = feature["type"];
             EXPECT_TRUE(type.IsInt());
             tileType = TileFeatureType(type.GetInt());
         }
+        */
+
+        mapbox::geometry::feature<short> feat { mapbox::geometry::point<short>() };
 
-        std::map<std::string, std::string> tileTags;
         if (feature.HasMember("tags") && feature["tags"].IsObject()) {
             const auto& tags = feature["tags"];
             for (auto jt = tags.MemberBegin(); jt != tags.MemberEnd(); jt++) {
                 const std::string tagKey{ jt->name.GetString(), jt->name.GetStringLength() };
                 switch (jt->value.GetType()) {
                 case rapidjson::kNullType:
-                    tileTags.emplace(tagKey, "null");
+                    feat.properties.emplace(tagKey, nullptr);
                     break;
                 case rapidjson::kFalseType:
-                    tileTags.emplace(tagKey, "false");
+                    feat.properties.emplace(tagKey, false);
                     break;
                 case rapidjson::kTrueType:
-                    tileTags.emplace(tagKey, "true");
+                    feat.properties.emplace(tagKey, true);
                     break;
                 case rapidjson::kStringType:
-                    tileTags.emplace(
-                        tagKey, std::string{ jt->value.GetString(), jt->value.GetStringLength() });
+                    feat.properties.emplace(tagKey, std::string{ jt->value.GetString(), jt->value.GetStringLength() });
                     break;
                 case rapidjson::kNumberType:
-                    tileTags.emplace(tagKey, std::to_string(jt->value.GetDouble()));
+
+                    // Somehow the features generated by geojson-cpp is
+                    // considering the `density` property of New Jersey (1189)
+                    // to be an int64_t rather than uint64_t.
+                    // https://github.com/mapbox/geojson-cpp/blob/master/src/mapbox/geojson.cpp#L121-L126
+                    //
+                    // if (jt->value.IsUint64()) {
+                    //     feat.properties.emplace(tagKey, std::uint64_t(jt->value.GetUint64()));
+                    // }
```

Comment:
:sweat: the New Jersey feature that's going through

```
mapbox::geojson::parse(loadFile("test/fixtures/us-states.json"))
```

has a density value that is `int64_t` because `geojson-cpp` is currently based on `geometry.hpp@0.4.0` -- which preferred int64_t to uint64_t:

https://github.com/mapbox/geometry.hpp/blob/d7a28b485b1e3ac25810e404cf41dac7b364cad1/include/mapbox/geometry/feature.hpp#L16